cpp/src/arrow/status.cc

```diff
@@ -131,6 +131,13 @@ void Status::Abort(const std::string& message) const {
   std::abort();
 }
 
+void Status::Warn() const { Warn(std::string()); }
+
+void Status::Warn(const std::string& message) const {
+  ARROW_LOG(WARNING) << "Arrow Warning -- "
+                     << message << ": " << ToString();
```

Comment:
If message is empty (which is the case when calling `Warn()` without arguments, this will leave a spurious colon.

Reply:
good point, thanks!